diff -up ffmpeg-20100425/configure.x264 ffmpeg-20100425/configure
--- ffmpeg-20100425/configure.x264	2010-04-23 18:41:55.000000000 +0200
+++ ffmpeg-20100425/configure	2010-04-25 23:31:45.000000000 +0200
@@ -2615,8 +2615,8 @@ enabled libspeex   && require  libspeex 
 enabled libtheora  && require  libtheora theora/theoraenc.h th_info_init -ltheoraenc -ltheoradec -logg
 enabled libvorbis  && require  libvorbis vorbis/vorbisenc.h vorbis_info_init -lvorbisenc -lvorbis -logg
 enabled libx264    && require  libx264 x264.h x264_encoder_encode -lx264 -lm &&
-                      { check_cpp_condition x264.h "X264_BUILD >= 83" ||
-                        die "ERROR: libx264 version must be >= 0.83."; }
+                      { check_cpp_condition x264.h "X264_BUILD >= 78" ||
+                        die "ERROR: libx264 version must be >= 0.78."; }
 enabled libxvid    && require  libxvid xvid.h xvid_global -lxvidcore
 enabled mlib       && require  mediaLib mlib_types.h mlib_VectorSub_S16_U8_Mod -lmlib
 
diff -up ffmpeg-20100425/ffpresets/libx264-baseline.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-baseline.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-baseline.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-baseline.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -1,4 +1,3 @@
 coder=0
 bf=0
 flags2=-wpred-dct8x8
-wpredp=0
diff -up ffmpeg-20100425/ffpresets/libx264-default.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-default.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-default.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-default.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -19,4 +19,3 @@ refs=3
 directpred=1
 trellis=1
 flags2=+mixed_refs+wpred+dct8x8+fastpskip
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-faster.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-faster.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-faster.ffpreset.x264	2010-03-29 03:34:33.000000000 +0200
+++ ffmpeg-20100425/ffpresets/libx264-faster.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4+partp8x8+partb8x8
-me_method=hex
-subq=4
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=2
-directpred=1
-trellis=1
-flags2=+bpyramid-mixed_refs+wpred+dct8x8+fastpskip
-wpredp=1
-rc_lookahead=20
diff -up ffmpeg-20100425/ffpresets/libx264-faster_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-faster_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-faster_firstpass.ffpreset.x264	2010-03-29 03:34:33.000000000 +0200
+++ ffmpeg-20100425/ffpresets/libx264-faster_firstpass.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=2
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=1
-directpred=1
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred-dct8x8+fastpskip
-wpredp=1
-rc_lookahead=20
diff -up ffmpeg-20100425/ffpresets/libx264-fast.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-fast.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-fast.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-fast.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4+partp8x8+partb8x8
-me_method=hex
-subq=6
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=2
-directpred=1
-trellis=1
-flags2=+bpyramid+mixed_refs+wpred+dct8x8+fastpskip
-wpredp=2
-rc_lookahead=30
diff -up ffmpeg-20100425/ffpresets/libx264-fast_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-fast_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-fast_firstpass.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-fast_firstpass.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=2
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=1
-directpred=1
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred-dct8x8+fastpskip
-wpredp=2
-rc_lookahead=30
diff -up ffmpeg-20100425/ffpresets/libx264-fastfirstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-fastfirstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-fastfirstpass.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-fastfirstpass.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -3,7 +3,7 @@ flags=+loop
 cmp=+chroma
 partitions=-parti8x8-parti4x4-partp8x8-partp4x4-partb8x8
 me_method=dia
-subq=2
+subq=1
 me_range=16
 g=250
 keyint_min=25
@@ -19,4 +19,3 @@ refs=1
 directpred=3
 trellis=0
 flags2=-bpyramid-wpred-mixed_refs-dct8x8+fastpskip
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-hq.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-hq.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-hq.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-hq.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -19,4 +19,3 @@ refs=4
 directpred=3
 trellis=1
 flags2=+wpred+mixed_refs+dct8x8+fastpskip
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-ipod320.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-ipod320.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-ipod320.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-ipod320.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -4,4 +4,3 @@ flags2=-wpred-dct8x8
 level=13
 maxrate=768000
 bufsize=3000000
-wpredp=0
diff -up ffmpeg-20100425/ffpresets/libx264-ipod640.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-ipod640.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-ipod640.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-ipod640.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -5,4 +5,3 @@ flags2=-wpred-dct8x8
 level=30
 maxrate=10000000
 bufsize=10000000
-wpredp=0
diff -up ffmpeg-20100425/ffpresets/libx264-lossless_fast.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-lossless_fast.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-lossless_fast.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-lossless_fast.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -17,4 +17,3 @@ qdiff=4
 directpred=1
 flags2=+fastpskip
 cqp=0
-wpredp=0
diff -up ffmpeg-20100425/ffpresets/libx264-lossless_max.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-lossless_max.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-lossless_max.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-lossless_max.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -18,4 +18,3 @@ refs=16
 directpred=1
 flags2=+mixed_refs+dct8x8+fastpskip
 cqp=0
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-lossless_medium.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-lossless_medium.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-lossless_medium.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-lossless_medium.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -17,4 +17,3 @@ qdiff=4
 directpred=1
 flags2=+fastpskip
 cqp=0
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-lossless_slower.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-lossless_slower.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-lossless_slower.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-lossless_slower.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -18,4 +18,3 @@ refs=4
 directpred=1
 flags2=+mixed_refs+dct8x8+fastpskip
 cqp=0
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-lossless_slow.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-lossless_slow.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-lossless_slow.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-lossless_slow.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -18,4 +18,3 @@ refs=2
 directpred=1
 flags2=+dct8x8+fastpskip
 cqp=0
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-lossless_ultrafast.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-lossless_ultrafast.ffpreset
diff -up ffmpeg-20100425/ffpresets/libx264-main.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-main.ffpreset
diff -up ffmpeg-20100425/ffpresets/libx264-max.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-max.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-max.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-max.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -19,4 +19,3 @@ refs=16
 directpred=3
 trellis=2
 flags2=+wpred+mixed_refs+dct8x8-fastpskip
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-medium.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-medium.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-medium.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-medium.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,22 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4+partp8x8+partb8x8
-me_method=hex
-subq=7
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=3
-directpred=1
-trellis=1
-flags2=+bpyramid+mixed_refs+wpred+dct8x8+fastpskip
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-medium_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-medium_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-medium_firstpass.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-medium_firstpass.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,22 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=2
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=1
-directpred=1
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred-dct8x8+fastpskip
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-normal.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-normal.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-normal.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-normal.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -19,4 +19,3 @@ refs=2
 directpred=3
 trellis=0
 flags2=+wpred+dct8x8+fastpskip
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-placebo.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-placebo.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-placebo.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-placebo.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4+partp8x8+partp4x4+partb8x8
-me_method=tesa
-subq=10
-me_range=24
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=2
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=16
-refs=16
-directpred=3
-trellis=2
-flags2=+bpyramid+mixed_refs+wpred+dct8x8-fastpskip
-wpredp=2
-rc_lookahead=60
diff -up ffmpeg-20100425/ffpresets/libx264-placebo_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-placebo_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-placebo_firstpass.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-placebo_firstpass.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4+partp8x8+partp4x4+partb8x8
-me_method=tesa
-subq=10
-me_range=24
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=2
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=16
-refs=16
-directpred=3
-trellis=2
-flags2=+bpyramid+mixed_refs+wpred+dct8x8-fastpskip
-wpredp=2
-rc_lookahead=60
diff -up ffmpeg-20100425/ffpresets/libx264-slower.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-slower.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-slower.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-slower.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4+partp8x8+partp4x4+partb8x8
-me_method=umh
-subq=9
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=2
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=8
-directpred=3
-trellis=2
-flags2=+bpyramid+mixed_refs+wpred+dct8x8+fastpskip
-wpredp=2
-rc_lookahead=60
diff -up ffmpeg-20100425/ffpresets/libx264-slower_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-slower_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-slower_firstpass.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-slower_firstpass.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=2
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=2
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=1
-directpred=3
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred-dct8x8+fastpskip
-wpredp=2
-rc_lookahead=60
diff -up ffmpeg-20100425/ffpresets/libx264-slow.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-slow.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-slow.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-slow.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4+partp8x8+partb8x8
-me_method=umh
-subq=8
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=2
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=5
-directpred=3
-trellis=1
-flags2=+bpyramid+mixed_refs+wpred+dct8x8+fastpskip
-wpredp=2
-rc_lookahead=50
diff -up ffmpeg-20100425/ffpresets/libx264-slow_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-slow_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-slow_firstpass.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-slow_firstpass.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=2
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=2
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=1
-directpred=3
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred-dct8x8+fastpskip
-wpredp=2
-rc_lookahead=50
diff -up ffmpeg-20100425/ffpresets/libx264-slowfirstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-slowfirstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-slowfirstpass.ffpreset.x264	2010-03-25 20:39:06.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-slowfirstpass.ffpreset	2010-04-25 23:31:45.000000000 +0200
@@ -19,4 +19,3 @@ refs=1
 directpred=3
 trellis=0
 flags2=+wpred+dct8x8+fastpskip
-wpredp=2
diff -up ffmpeg-20100425/ffpresets/libx264-superfast.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-superfast.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-superfast.ffpreset.x264	2010-03-29 03:34:33.000000000 +0200
+++ ffmpeg-20100425/ffpresets/libx264-superfast.ffpreset	2010-04-25 23:28:45.000000000 +0200
@@ -1,22 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=1
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=1
-directpred=1
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred+dct8x8+fastpskip-mbtree
-wpredp=0
diff -up ffmpeg-20100425/ffpresets/libx264-superfast_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-superfast_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-superfast_firstpass.ffpreset.x264	2010-03-29 03:34:33.000000000 +0200
+++ ffmpeg-20100425/ffpresets/libx264-superfast_firstpass.ffpreset	2010-04-25 23:28:45.000000000 +0200
@@ -1,22 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=1
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=1
-directpred=1
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred-dct8x8+fastpskip-mbtree
-wpredp=0
diff -up ffmpeg-20100425/ffpresets/libx264-ultrafast.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-ultrafast.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-ultrafast.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-ultrafast.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=0
-flags=-loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=0
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=0
-i_qfactor=0.71
-b_strategy=0
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=0
-refs=1
-directpred=1
-trellis=0
-flags2=-bpyramid-mixed_refs-wpred-dct8x8+fastpskip-mbtree
-wpredp=0
-aq_mode=0
diff -up ffmpeg-20100425/ffpresets/libx264-ultrafast_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-ultrafast_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-ultrafast_firstpass.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-ultrafast_firstpass.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=0
-flags=-loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=0
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=0
-i_qfactor=0.71
-b_strategy=0
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=0
-refs=1
-directpred=1
-trellis=0
-flags2=-bpyramid-mixed_refs-wpred-dct8x8+fastpskip-mbtree
-wpredp=0
-aq_mode=0
diff -up ffmpeg-20100425/ffpresets/libx264-veryfast.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-veryfast.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-veryfast.ffpreset.x264	2010-03-29 03:34:33.000000000 +0200
+++ ffmpeg-20100425/ffpresets/libx264-veryfast.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,22 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4+partp8x8+partb8x8
-me_method=hex
-subq=2
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=1
-directpred=1
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred+dct8x8+fastpskip-mbtree
-wpredp=0
diff -up ffmpeg-20100425/ffpresets/libx264-veryfast_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-veryfast_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-veryfast_firstpass.ffpreset.x264	2010-03-29 03:34:33.000000000 +0200
+++ ffmpeg-20100425/ffpresets/libx264-veryfast_firstpass.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,22 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=2
-me_range=16
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=1
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=3
-refs=1
-directpred=1
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred-dct8x8+fastpskip-mbtree
-wpredp=0
diff -up ffmpeg-20100425/ffpresets/libx264-veryslow.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-veryslow.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-veryslow.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-veryslow.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=+parti8x8+parti4x4+partp8x8+partp4x4+partb8x8
-me_method=umh
-subq=10
-me_range=24
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=2
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=8
-refs=16
-directpred=3
-trellis=2
-flags2=+bpyramid+mixed_refs+wpred+dct8x8+fastpskip
-wpredp=2
-rc_lookahead=60
diff -up ffmpeg-20100425/ffpresets/libx264-veryslow_firstpass.ffpreset.x264 ffmpeg-20100425/ffpresets/libx264-veryslow_firstpass.ffpreset
--- ffmpeg-20100425/ffpresets/libx264-veryslow_firstpass.ffpreset.x264	2010-03-26 23:40:55.000000000 +0100
+++ ffmpeg-20100425/ffpresets/libx264-veryslow_firstpass.ffpreset	2010-04-25 23:30:58.000000000 +0200
@@ -1,23 +0,0 @@
-coder=1
-flags=+loop
-cmp=+chroma
-partitions=-parti8x8-parti4x4-partp8x8-partb8x8
-me_method=dia
-subq=2
-me_range=24
-g=250
-keyint_min=25
-sc_threshold=40
-i_qfactor=0.71
-b_strategy=2
-qcomp=0.6
-qmin=10
-qmax=51
-qdiff=4
-bf=8
-refs=1
-directpred=3
-trellis=0
-flags2=+bpyramid-mixed_refs+wpred-dct8x8+fastpskip
-wpredp=2
-rc_lookahead=60
diff -up ffmpeg-20100425/libavcodec/avcodec.h.x264 ffmpeg-20100425/libavcodec/avcodec.h
--- ffmpeg-20100425/libavcodec/avcodec.h.x264	2010-04-20 16:45:34.000000000 +0200
+++ ffmpeg-20100425/libavcodec/avcodec.h	2010-04-25 23:31:45.000000000 +0200
@@ -594,9 +594,6 @@ typedef struct RcOverride{
 #define CODEC_FLAG2_CHUNKS        0x00008000 ///< Input bitstream might be truncated at a packet boundaries instead of only at frame boundaries.
 #define CODEC_FLAG2_NON_LINEAR_QUANT 0x00010000 ///< Use MPEG-2 nonlinear quantizer.
 #define CODEC_FLAG2_BIT_RESERVOIR 0x00020000 ///< Use a bit reservoir when encoding if possible
-#define CODEC_FLAG2_MBTREE        0x00040000 ///< Use macroblock tree ratecontrol (x264 only)
-#define CODEC_FLAG2_PSY           0x00080000 ///< Use psycho visual optimizations.
-#define CODEC_FLAG2_SSIM          0x00100000 ///< Compute SSIM during encoding, error[] values are undefined.
 
 /* Unsupported options :
  *              Syntax Arithmetic coding (SAC)
@@ -2594,58 +2591,6 @@ typedef struct AVCodecContext {
      * - decoding: Set by libavcodec, user can override.
      */
     int (*execute2)(struct AVCodecContext *c, int (*func)(struct AVCodecContext *c2, void *arg, int jobnr, int threadnr), void *arg2, int *ret, int count);
-
-    /**
-     * explicit P-frame weighted prediction analysis method
-     * 0: off
-     * 1: fast blind weighting (one reference duplicate with -1 offset)
-     * 2: smart weighting (full fade detection analysis)
-     * - encoding: Set by user.
-     * - decoding: unused
-     */
-    int weighted_p_pred;
-
-    /**
-     * AQ mode
-     * 0: Disabled
-     * 1: Variance AQ (complexity mask)
-     * 2: Auto-variance AQ (experimental)
-     * - encoding: Set by user
-     * - decoding: unused
-     */
-    int aq_mode;
-
-    /**
-     * AQ strength
-     * Reduces blocking and blurring in flat and textured areas.
-     * - encoding: Set by user
-     * - decoding: unused
-     */
-    float aq_strength;
-
-    /**
-     * PSY RD
-     * Strength of psychovisual optimization
-     * - encoding: Set by user
-     * - decoding: unused
-     */
-    float psy_rd;
-
-    /**
-     * PSY trellis
-     * Strength of psychovisual optimization
-     * - encoding: Set by user
-     * - decoding: unused
-     */
-    float psy_trellis;
-
-    /**
-     * RC lookahead
-     * Number of frames for frametype and ratecontrol lookahead
-     * - encoding: Set by user
-     * - decoding: unused
-     */
-    int rc_lookahead;
 } AVCodecContext;
 
 /**
diff -up ffmpeg-20100425/libavcodec/libx264.c.x264 ffmpeg-20100425/libavcodec/libx264.c
--- ffmpeg-20100425/libavcodec/libx264.c.x264	2010-04-19 23:20:20.000000000 +0200
+++ ffmpeg-20100425/libavcodec/libx264.c	2010-04-25 23:31:45.000000000 +0200
@@ -109,7 +109,7 @@ static int X264_frame(AVCodecContext *ct
     if (bufsize < 0)
         return -1;
 
-    /* FIXME: libx264 now provides DTS, but AVFrame doesn't have a field for it. */
+    /* FIXME: dts */
     x4->out_pic.pts = pic_out.i_pts;
 
     switch (pic_out.i_type) {
@@ -126,7 +126,7 @@ static int X264_frame(AVCodecContext *ct
         break;
     }
 
-    x4->out_pic.key_frame = pic_out.b_keyframe;
+    x4->out_pic.key_frame = pic_out.i_type == X264_TYPE_IDR;
     x4->out_pic.quality   = (pic_out.i_qpplus1 - 1) * FF_QP2LAMBDA;
 
     return bufsize;
@@ -208,8 +208,8 @@ static av_cold int X264_init(AVCodecCont
     x4->params.i_height             = avctx->height;
     x4->params.vui.i_sar_width      = avctx->sample_aspect_ratio.num;
     x4->params.vui.i_sar_height     = avctx->sample_aspect_ratio.den;
-    x4->params.i_fps_num = x4->params.i_timebase_den = avctx->time_base.den;
-    x4->params.i_fps_den = x4->params.i_timebase_num = avctx->time_base.num;
+    x4->params.i_fps_num            = avctx->time_base.den;
+    x4->params.i_fps_den            = avctx->time_base.num;
 
     x4->params.analyse.inter    = 0;
     if (avctx->partitions) {
@@ -228,7 +228,6 @@ static av_cold int X264_init(AVCodecCont
     x4->params.analyse.i_direct_mv_pred  = avctx->directpred;
 
     x4->params.analyse.b_weighted_bipred = avctx->flags2 & CODEC_FLAG2_WPRED;
-    x4->params.analyse.i_weighted_pred = avctx->weighted_p_pred;
 
     if (avctx->me_method == ME_EPZS)
         x4->params.analyse.i_me_method = X264_ME_DIA;
@@ -242,14 +241,6 @@ static av_cold int X264_init(AVCodecCont
         x4->params.analyse.i_me_method = X264_ME_TESA;
     else x4->params.analyse.i_me_method = X264_ME_HEX;
 
-    x4->params.rc.i_aq_mode               = avctx->aq_mode;
-    x4->params.rc.f_aq_strength           = avctx->aq_strength;
-    x4->params.rc.i_lookahead             = avctx->rc_lookahead;
-
-    x4->params.analyse.b_psy              = avctx->flags2 & CODEC_FLAG2_PSY;
-    x4->params.analyse.f_psy_rd           = avctx->psy_rd;
-    x4->params.analyse.f_psy_trellis      = avctx->psy_trellis;
-
     x4->params.analyse.i_me_range         = avctx->me_range;
     x4->params.analyse.i_subpel_refine    = avctx->me_subpel_quality;
 
@@ -274,13 +265,11 @@ static av_cold int X264_init(AVCodecCont
     } else
         x4->params.rc.f_vbv_buffer_init = 0.9;
 
-    x4->params.rc.b_mb_tree               = !!(avctx->flags2 & CODEC_FLAG2_MBTREE);
     x4->params.rc.f_ip_factor             = 1 / fabs(avctx->i_quant_factor);
     x4->params.rc.f_pb_factor             = avctx->b_quant_factor;
     x4->params.analyse.i_chroma_qp_offset = avctx->chromaoffset;
 
     x4->params.analyse.b_psnr = avctx->flags & CODEC_FLAG_PSNR;
-    x4->params.analyse.b_ssim = avctx->flags2 & CODEC_FLAG2_SSIM;
     x4->params.i_log_level    = X264_LOG_DEBUG;
 
     x4->params.b_aud          = avctx->flags2 & CODEC_FLAG2_AUD;
@@ -300,14 +289,10 @@ static av_cold int X264_init(AVCodecCont
 
     if (avctx->flags & CODEC_FLAG_GLOBAL_HEADER) {
         x264_nal_t *nal;
-        int nnal, s, i;
+        int nnal, s;
 
         s = x264_encoder_headers(x4->enc, &nal, &nnal);
 
-        for (i = 0; i < nnal; i++)
-            if (nal[i].i_type == NAL_SEI)
-                av_log(avctx, AV_LOG_INFO, "%s\n", nal[i].p_payload+25);
-
         avctx->extradata      = av_malloc(s);
         avctx->extradata_size = encode_nals(avctx, avctx->extradata, s, nal, nnal, 1);
     }
diff -up ffmpeg-20100425/libavcodec/options.c.x264 ffmpeg-20100425/libavcodec/options.c
--- ffmpeg-20100425/libavcodec/options.c.x264	2010-04-20 16:45:34.000000000 +0200
+++ ffmpeg-20100425/libavcodec/options.c	2010-04-25 23:31:45.000000000 +0200
@@ -124,7 +124,6 @@ static const AVOption options[]={
 {"b_qfactor", "qp factor between p and b frames", OFFSET(b_quant_factor), FF_OPT_TYPE_FLOAT, 1.25, -FLT_MAX, FLT_MAX, V|E},
 {"rc_strategy", "ratecontrol method", OFFSET(rc_strategy), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX, V|E},
 {"b_strategy", "strategy to choose between I/P/B-frames", OFFSET(b_frame_strategy), FF_OPT_TYPE_INT, 0, INT_MIN, INT_MAX, V|E},
-{"wpredp", "weighted prediction analysis method", OFFSET(weighted_p_pred), FF_OPT_TYPE_INT, 0, INT_MIN, INT_MAX, V|E},
 {"hurry_up", NULL, OFFSET(hurry_up), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX, V|D},
 {"ps", "rtp payload size in bytes", OFFSET(rtp_payload_size), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX, V|E},
 {"mv_bits", NULL, OFFSET(mv_bits), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX},
@@ -307,7 +306,7 @@ static const AVOption options[]={
 {"nr", "noise reduction", OFFSET(noise_reduction), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX, V|E},
 {"rc_init_occupancy", "number of bits which should be loaded into the rc buffer before decoding starts", OFFSET(rc_initial_buffer_occupancy), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX, V|E},
 {"inter_threshold", NULL, OFFSET(inter_threshold), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX, V|E},
-{"flags2", NULL, OFFSET(flags2), FF_OPT_TYPE_FLAGS, CODEC_FLAG2_FASTPSKIP|CODEC_FLAG2_BIT_RESERVOIR|CODEC_FLAG2_PSY|CODEC_FLAG2_MBTREE, 0, UINT_MAX, V|A|E|D, "flags2"},
+{"flags2", NULL, OFFSET(flags2), FF_OPT_TYPE_FLAGS, CODEC_FLAG2_FASTPSKIP|CODEC_FLAG2_BIT_RESERVOIR, 0, UINT_MAX, V|A|E|D, "flags2"},
 {"error", NULL, OFFSET(error_rate), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX, V|E},
 {"antialias", "MP3 antialias algorithm", OFFSET(antialias_algo), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX, V|D, "aa"},
 {"auto", NULL, 0, FF_OPT_TYPE_CONST, FF_AA_AUTO, INT_MIN, INT_MAX, V|D, "aa"},
@@ -392,7 +391,6 @@ static const AVOption options[]={
 {"request_channels", "set desired number of audio channels", OFFSET(request_channels), FF_OPT_TYPE_INT, DEFAULT, 0, INT_MAX, A|D},
 {"drc_scale", "percentage of dynamic range compression to apply", OFFSET(drc_scale), FF_OPT_TYPE_FLOAT, 1.0, 0.0, 1.0, A|D},
 {"reservoir", "use bit reservoir", 0, FF_OPT_TYPE_CONST, CODEC_FLAG2_BIT_RESERVOIR, INT_MIN, INT_MAX, A|E, "flags2"},
-{"mbtree", "use macroblock tree ratecontrol (x264 only)", 0, FF_OPT_TYPE_CONST, CODEC_FLAG2_MBTREE, INT_MIN, INT_MAX, V|E, "flags2"},
 {"bits_per_raw_sample", NULL, OFFSET(bits_per_raw_sample), FF_OPT_TYPE_INT, DEFAULT, INT_MIN, INT_MAX},
 {"channel_layout", NULL, OFFSET(channel_layout), FF_OPT_TYPE_INT64, DEFAULT, 0, INT64_MAX, A|E|D, "channel_layout"},
 {"request_channel_layout", NULL, OFFSET(request_channel_layout), FF_OPT_TYPE_INT64, DEFAULT, 0, INT64_MAX, A|D, "request_channel_layout"},
@@ -404,13 +402,6 @@ static const AVOption options[]={
 {"colorspace", NULL, OFFSET(colorspace), FF_OPT_TYPE_INT, AVCOL_SPC_UNSPECIFIED, 1, AVCOL_SPC_NB-1, V|E|D},
 {"color_range", NULL, OFFSET(color_range), FF_OPT_TYPE_INT, AVCOL_RANGE_UNSPECIFIED, 0, AVCOL_RANGE_NB-1, V|E|D},
 {"chroma_sample_location", NULL, OFFSET(chroma_sample_location), FF_OPT_TYPE_INT, AVCHROMA_LOC_UNSPECIFIED, 0, AVCHROMA_LOC_NB-1, V|E|D},
-{"psy", "use psycho visual optimization", 0, FF_OPT_TYPE_CONST, CODEC_FLAG2_PSY, INT_MIN, INT_MAX, V|E, "flags2"},
-{"psy_rd", "specify psycho visual strength", OFFSET(psy_rd), FF_OPT_TYPE_FLOAT, 1.0, 0, FLT_MAX, V|E},
-{"psy_trellis", "specify psycho visual trellis", OFFSET(psy_trellis), FF_OPT_TYPE_FLOAT, 0, 0, FLT_MAX, V|E},
-{"aq_mode", "specify aq method", OFFSET(aq_mode), FF_OPT_TYPE_INT, 1, 0, INT_MAX, V|E},
-{"aq_strength", "specify aq strength", OFFSET(aq_strength), FF_OPT_TYPE_FLOAT, 1.0, 0, FLT_MAX, V|E},
-{"rc_lookahead", "specify number of frames to look ahead for frametype", OFFSET(rc_lookahead), FF_OPT_TYPE_INT, 40, 0, INT_MAX, V|E},
-{"ssim", "ssim will be calculated during encoding", 0, FF_OPT_TYPE_CONST, CODEC_FLAG2_SSIM, INT_MIN, INT_MAX, V|E, "flags2"},
 {NULL},
 };
 
