diff -up ffmpeg-20080908/libavformat/matroska.c.drc ffmpeg-20080908/libavformat/matroska.c
--- ffmpeg-20080908/libavformat/matroska.c.drc	2008-09-01 17:50:12.000000000 +0200
+++ ffmpeg-20080908/libavformat/matroska.c	2009-02-09 21:09:21.000000000 +0100
@@ -23,6 +23,7 @@
 
 const CodecTags ff_mkv_codec_tags[]={
     {"V_UNCOMPRESSED"   , CODEC_ID_RAWVIDEO},
+    {"V_DIRAC"          , CODEC_ID_DIRAC},
     {"V_MPEG4/ISO/ASP"  , CODEC_ID_MPEG4},
     {"V_MPEG4/ISO/SP"   , CODEC_ID_MPEG4},
     {"V_MPEG4/ISO/AP"   , CODEC_ID_MPEG4},
