diff -up ffmpeg-20090204/libavcodec/i386/fft_mmx.asm.textrel ffmpeg-20090204/libavcodec/i386/fft_mmx.asm
--- ffmpeg-20090204/libavcodec/i386/fft_mmx.asm.textrel	2008-12-19 04:01:08.000000000 +0100
+++ ffmpeg-20090204/libavcodec/i386/fft_mmx.asm	2009-02-08 18:51:15.000000000 +0100
@@ -446,8 +446,9 @@ fft %+ n %+ %3%2:
 %endrep
 %undef n
 
-align 8
+section .data
 dispatch_tab%3%2: pointer list_of_fft
+section .text
 
 ; On x86_32, this function does the register saving and restoring for all of fft.
 ; The others pass args in registers and don't spill anything.
