diff -up ffmpeg-20080614/libavcodec/Makefile.asmreg ffmpeg-20080614/libavcodec/Makefile
--- ffmpeg-20080614/libavcodec/Makefile.asmreg	2008-06-08 23:08:57.000000000 +0200
+++ ffmpeg-20080614/libavcodec/Makefile	2008-06-14 17:24:58.000000000 +0200
@@ -87,6 +87,11 @@ OBJS-$(CONFIG_H263I_DECODER)           +
 OBJS-$(CONFIG_H263_ENCODER)            += mpegvideo_enc.o motion_est.o ratecontrol.o h263.o mpeg12data.o mpegvideo.o error_resilience.o
 OBJS-$(CONFIG_H263P_ENCODER)           += mpegvideo_enc.o motion_est.o ratecontrol.o h263.o mpeg12data.o mpegvideo.o error_resilience.o
 OBJS-$(CONFIG_H264_DECODER)            += h264.o h264idct.o h264pred.o h264_parser.o cabac.o golomb.o mpegvideo.o error_resilience.o
+# to "fix" can't find a register in class 'GENERAL_REGS' while reloading 'asm'
+# on ix86 in cabac.h
+ifeq ($(ARCH_X86_32),yes)
+h264.o: CFLAGS += -fomit-frame-pointer
+endif
 OBJS-$(CONFIG_H264_ENCODER)            += h264enc.o h264dspenc.o
 OBJS-$(CONFIG_HUFFYUV_DECODER)         += huffyuv.o
 OBJS-$(CONFIG_HUFFYUV_ENCODER)         += huffyuv.o
@@ -391,6 +396,11 @@ OBJS-$(CONFIG_ENCODERS)                +
 OBJS-$(CONFIG_CAVS_DECODER)            += i386/cavsdsp_mmx.o
 OBJS-$(CONFIG_FLAC_ENCODER)            += i386/flacdsp_mmx.o
 OBJS-$(CONFIG_SNOW_DECODER)            += i386/snowdsp_mmx.o
+# to "fix" can't find a register in class 'GENERAL_REGS' while reloading 'asm'
+# on ix86 in snowdsp_mmx.c
+ifeq ($(ARCH_X86_32),yes)
+i386/snowdsp_mmx.o: CFLAGS += -fomit-frame-pointer
+endif
 OBJS-$(CONFIG_VC1_DECODER)             += i386/vc1dsp_mmx.o
 OBJS-$(CONFIG_VP3_DECODER)             += i386/vp3dsp_mmx.o i386/vp3dsp_sse2.o
 OBJS-$(CONFIG_VP5_DECODER)             += i386/vp3dsp_mmx.o i386/vp3dsp_sse2.o
