diff -up ffmpeg-20091026/configure~ ffmpeg-20091026/configure
--- ffmpeg-20091026/configure~	2009-10-24 19:24:25.000000000 +0200
+++ ffmpeg-20091026/configure	2009-11-07 16:35:15.000000000 +0100
@@ -1805,12 +1805,14 @@ case "$arch" in
     "Power Macintosh"|ppc|powerpc)
         arch="ppc"
         enable fast_unaligned
+        spic=$shared
     ;;
     ppc64)
         arch="ppc"
         subarch="ppc64"
         enable fast_64bit
         enable fast_unaligned
+        spic=$shared
     ;;
     s390|s390x)
         arch="s390"
